sources/packages/api/src/services/student/student.service.ts

```diff
@@ -176,6 +176,20 @@ export class StudentService extends RecordDataModelService<Student> {
     return this.repo.find({ validSIN: null });
   }
 
+  /**
+   * Gets a student's valid sin status
+   * @returns Students SIN validation status.
+   */
+  async getStudentSinStatus(userId: number): Promise<boolean | null> {
+    const sinStatus = await this.repo
+      .createQueryBuilder("student")
+      .leftJoinAndSelect("student.user", "user")
+      .where("student.userId = :userIdParam", { userIdParam: userId })
+      .select("student.validSin")
+      .getOne();
+    return sinStatus.validSIN;
```

Comment:
actually, if you check other service methods, you can see that we are following like, service will be returning the DB entity as such, and on consumer side, the needed data is retrieved. we are doing like so that we can reuse the service layer

Reply:
Correct, there is already a service that returns Studentinfo entity currently used on the calling side. This one was created specifically for the decorator to check validSIN status.